### PR TITLE
Simple UDP Example notebook

### PR DESCRIPTION
**Pull request by Pratichhya**

### CODE REVIEW

File: openeo/UDP.ipynb [hunk 1]
```diff
@@ -0,0 +1,278 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# User-Defined Processes(UDP)\n",
```
Comment (soxofaan) [inline openeo/UDP.ipynb:8]:
```suggestion
    "# User-Defined Processes (UDP)\n",
```

File: openeo/UDP.ipynb [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define input parameters\n",
+    "time_param = Parameter(\n",
+    "                            name = 'date',\n",
+    "                            description = \"A date within wich you wish to load the Terrasope NDVI product.\",\n",
+    "                            schema = {\"type\":\"array\",\"subtype\":\"data-time\"}\n",
```
Comment (soxofaan) [inline openeo/UDP.ipynb:65]: I'm confused here: `time_param` looks like  single date so must be a string I think, but here it's `"type":"array"`. 

Also `"subtype":"data-time"` look invalid (e.g. `data-time` is not a thing)

**Review by soxofaan — COMMENTED** [review]
a couple of notes

**Comment by Pratichhya** [comment]
> Looks quite good, not much to add! Showing the json generated by flat_graph is maybe not needed, I always find it fairly difficult, and is not crucial. I think there's also a way that shows a more graphical rendering of the graph. Maybe @soxofaan knows how?

Adopted the solution suggested by @soxofaan

File: openeo/UDP.ipynb (hunk 2 above)
Comment (Pratichhya) [reply openeo/UDP.ipynb:66]: > Another thing: I find this indentation approach very ugly and wasteful
> The conventional way is to just indent with 4 spaces

Updated

Comment (Pratichhya) [reply openeo/UDP.ipynb:66]: > I'm confused here: `time_param` looks like single date so must be a string I think, but here it's `"type":"array"`.
> 
> Also `"subtype":"data-time"` look invalid (e.g. `data-time` is not a thing)

Updated as string

File: openeo/UDP.ipynb [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+    "\n",
+    "ndwi = ((green - nir) / (green + nir))"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can also inspect its openEO-style process graph representation, use the `.flat_graph()` method."
```
[the remark this answers is not in the document]
Comment (Pratichhya) [reply openeo/UDP.ipynb:119]: updated

File: openeo/UDP.ipynb [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bbox = {\"west\":5.09, \"south\":51.18,\"east\":5.15,\"north\":51.21, \"crs\":4326}\n",
+    "check_process = connection.datacube_from_process(process_id=\"NDWI\", date = [\"2022-06-01\", \"2022-10-01\"],aoi=bbox)"
```
[the remark this answers is not in the document]
Comment (Pratichhya) [reply openeo/UDP.ipynb:238]: updated with a name created_process()

[the remark this answers is not in the document]
Comment (Pratichhya) [reply openeo/UDP.ipynb:238]: Also 4 indentation adopted

File: openeo/UDP.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (soxofaan) [reply openeo/UDP.ipynb:153]: > You can also inspect its openEO-style process graph representation, use the `.flat_graph()` method.

remove this whole line, we should not promote usage of `flat_graph` method

File: openeo/UDP.ipynb [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# publishing the service\n",
+    "process_name = \"NDWI\"\n",
+    "my_graph = ndwi.flat_graph()\n",
```
Comment (soxofaan) [inline openeo/UDP.ipynb:204]: `my_graph = ndwi.flat_graph()`

this is not necessary, you can directly pass `ndwi` to `save_user_defined_process`

File: openeo/UDP.ipynb [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "C:\\Users\\SHARMAP\\AppData\\Local\\Temp\\ipykernel_24856\\1719649594.py:9: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.\n",
```
Comment (soxofaan) [inline openeo/UDP.ipynb:294]: Can we avoid that this warning shows up? Looks unnecessary

**Comment by Pratichhya** [comment]
Used the suggested formater and updated the suggested changes.

**Comment by soxofaan** [comment]
looks much better now

I think time it's time to merge current state

**Review by soxofaan — APPROVED** [review]

**Pull request merged**